### PR TITLE
feat(cosmos): upgrade provisioning vat

### LINKED ISSUE #8113: can't send to account by address: namesByAddress is empty
```[tasklist]
### Tasks
- [ ] #8821
- [ ] https://github.com/Agoric/agoric-sdk/issues/8887
- [ ] https://github.com/Agoric/agoric-sdk/pull/8901
```
## Describe the bug

#5822 involves sending assets to accounts by address. As described in #8115, it should work something like...

```js
const destDepositFacet = E(home.namesByAddress).lookup(addr, 'depositFacet');
```

but  `E(namesByAddress).entries()` is empty: it resolves to `[]`.

@anilhelvaci reported seeing `[]` in a REPL. I reproduced it in an ava test.

## To Reproduce
1. `cd agoric-sdk/packages/vats`
2. apply diff below to the `sim/demo config provides home with .myAddressNameAdmin` test to show the entries
3. `yarn test test/bootstrapTests/test-demo-config.js -m '*home*'`
4. see an empty list of entries: `[]`

```diff
diff --git a/packages/vats/test/bootstrapTests/test-demo-config.js b/packages/vats/test/bootstrapTests/test-demo-config.js
index d9a53b5fd..41743b0c5 100644
--- a/packages/vats/test/bootstrapTests/test-demo-config.js
+++ b/packages/vats/test/bootstrapTests/test-demo-config.js
@@ -64,6 +64,8 @@ test('sim/demo config provides home with .myAddressNameAdmin', async t => {
   const home = await makeHomeFor(addr, EV);
   const actual = await EV(home.myAddressNameAdmin).getMyAddress();
   t.is(actual, addr, 'my address');
+  const xxx = await EV(home.namesByAddress).entries();
+  console.log('@@@@', xxx);
   keyArrayEqual(t, keys(home).sort(), homeKeys);
 });
```

## Expected behavior
An entry for each provisioned user.

## Platform Environment
 - _what OS are you using? what version of Node.js?_

```
$ uname -a
Linux bldbox 5.15.0-76-generic #83-Ubuntu SMP Thu Jun 15 19:16:32 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
$ node --version
v18.14.0
```

 - is there anything special/unusual about your platform?

I was working on a branch at the time I reproduced the problem, but it also reproduces on master.

 - what version of the Agoric-SDK are you using? (run `git describe --tags --always`)
`@agoric/cosmic-swingset@0.41.3-369-ge28392f88` e28392f88

## Additional context

@anilhelvaci reports that this is blocking a [bounty issue](https://github.com/Agoric/agoric-sdk/labels/bounty) that @schnetzlerjoe is working on (most likely #5822)

## Screenshots

```
$ yarn test test/bootstrapTests/test-demo-config.js -m '*home*'
...
@@@@ []
  ✔ sim/demo config provides home with .myAddressNameAdmin (16.2s)
    ℹ bootstrap produced provisioning vat
```

### PR DESCRIPTION
**Pull request by mhofman**
refs: #8821 #8311 #8786
closes: #8113 

Stacked on #8902

## Description

Core proposal to upgrade the provisioning vat (follow-up from #8821) and a3p integration test to verify namesByAddress works as expected (adapted from #8786)

### Security Considerations

Same as #8821

### Scaling Considerations

None

### Documentation Considerations

None

### Testing Considerations

Adds a3p integration test

### Upgrade Considerations

Core proposal to be executed during chain software upgrade. Tested in a3p. Has equivalent PR on the release branch.

### CODE REVIEW

**Comment by mhofman** [comment]
FYI @dckc

File: a3p-integration/proposals/a:upgrade-next/core-eval-support.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
+// @ts-check
```
Comment (dckc) [inline a3p-integration/proposals/a:upgrade-next/core-eval-support.js]: hm. another copy of this file? we can't get the relevant stuff from the synthetic chain package?

File: golang/cosmos/app/app.go [hunk 2]
```diff
@@ -849,6 +849,8 @@ func unreleasedUpgradeHandler(app *GaiaApp, targetUpgrade string) func(sdk.Conte
 			vm.CoreProposalStepForModules("@agoric/builders/scripts/smart-wallet/build-wallet-factory2-upgrade.js"),
 			// Then, upgrade Zoe and ZCF
 			vm.CoreProposalStepForModules("@agoric/builders/scripts/vats/replace-zoe.js"),
+			// Then, upgrade the provisioning vat
+			vm.CoreProposalStepForModules("@agoric/builders/scripts/vats/replace-provisioning.js"),
```
Comment (dckc) [inline golang/cosmos/app/app.go:853]: this is more than a test... ah... the commit message says `feat`. The PR title threw me off a bit.

File: a3p-integration/proposals/a:upgrade-next/yarn.lock [hunk 3]
```diff
@@ -335,6 +335,16 @@ __metadata:
   languageName: node
   linkType: hard
 
+"brace-expansion@npm:^1.1.7":
```
Comment (dckc) [inline a3p-integration/proposals/a:upgrade-next/yarn.lock]: all these changes to `yarn.lock` come from adding `tmp`?

I guess this is not our main `yarn.lock`, so there's not much supply-chain risks.

**Review by dckc — APPROVED** [review]
looks like this closes #8113!

File: a3p-integration/proposals/a:upgrade-next/core-eval-support.js (hunk 1 above)
Comment (mhofman) [reply a3p-integration/proposals/a:upgrade-next/core-eval-support.js]: If there's a way, I'm open to it. I just extracted what you had in https://github.com/Agoric/agoric-sdk/pull/8786. I'm open to alternatives!

**Comment by mhofman** [comment]
I will likely extract the core-eval-support part into a separate stand alone PR so that #8909 can stand on it too.

File: a3p-integration/proposals/a:upgrade-next/core-eval-support.js (hunk 1 above)
Comment (mhofman) [reply a3p-integration/proposals/a:upgrade-next/core-eval-support.js]: I think I found a way. PTAL

File: a3p-integration/proposals/a:upgrade-next/yarn.lock (hunk 3 above)
Comment (mhofman) [reply a3p-integration/proposals/a:upgrade-next/yarn.lock]: removed

**Review by dckc — APPROVED** [review]
even nicer

**Comment by mhofman** [comment]
Ran into a synthetic-chain bug: https://github.com/Agoric/agoric-3-proposals/pull/104

**Pull request merged**